woocommerce/woocommerce-android
# Issue/11196 widget editor screen

Closes: #11196


### Description

Ads new screen for adding/removing or reordering the available widget from My Store
In order to avoid massive UI code duplication, I've refactored some existing components used in `AnalyticsHubSettingScreen` into more general-purpose ones. 

### Testing instructions

1. Run the app and click on the new `Settings` CTA on the top right corner
2. Select unselect items
3. Reorder them
4. Clickin on the `X` button and check the confirmation dialog is shown
5. Clickin on `Save` button will crash the app as it is not yet implemented

**Analytics hub settings**
Smoke test analytics hub settings screen to ensure everything works as expected and there are no regressions after the refactor

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/2e564e6e-a585-48c3-a0e3-dfb464cf2cc7
